File: 2014-04-18-FP/JumpingBunnies/jb.accepted.fs
open System

let rec gcd a b =
        if b = 0UL
            then a
            else
                if a < b
                    then gcd b a
                    else gcd b (a % b)

let lcm a b = (a / (gcd a b)) * b

let main _ =
        ignore (Console.ReadLine ())
        let str = Console.ReadLine ()
        let xs = List.map uint64 (List.ofSeq (str.Split [|' '|]))
        Console.WriteLine ("{0}", List.fold lcm 1UL xs)

main ()

File: 2014-04-18-FP/JumpingBunnies/jb.accepted.scala
object Solution {
    @annotation.tailrec
    def gcd(a: BigInt, b: BigInt): BigInt =
        if (b == 0)
            a
        else if (a < b)
            gcd(b, a)
        else
            gcd(b, a % b)

    def lcm(a: BigInt, b: BigInt): BigInt = (a / gcd(a, b)) * b

    def main(args: Array[String]) = {
        readLine
        val xs = readLine.split(" ").map((x: String) => BigInt(x.toInt))
        println((xs foldLeft BigInt(1))(lcm))
    }
}
